### DISCUSSION TITLE
How to turn off some icons.

### DISCUSSION THREAD

**Discussion by ChrisTaylorDeveloper** [nvim-neo-tree/neo-tree.nvim#1597] [Q&A] [answered]
I would like to turn-off all icons, except for folder icons.
How can I do that please?

**Comment by pynappo** [comment, marked as answer]
You can modify how each row of the tree is rendered by modifying the renderers table for each source (filesystem, git_status, etc). Omitting a component will prevent neo-tree from rendering.

for example for filesystem:

https://github.com/nvim-neo-tree/neo-tree.nvim/blob/60c8f77cc35f9a52b7ee8da5df5555bd0e41b9f4/lua/neo-tree/defaults.lua#L305-L307

comment out the icon line there and now icons won't be rendered for files in the filesystem source. it should look like:

![image](https://github.com/user-attachments/assets/7965a460-37c1-49df-ac75-e768360f0c09)

**Reply by ChrisTaylorDeveloper** [reply]
Thank you.  Works perfectly when I comment out that line in `~/.local/share/nvim/lazy/neo-tree.nvim/lua/neo-tree/defaults.lua` but is there an easy way to make that change in my own settings?  Do I need to copy the whole `file = { ... }` section but leave out the `{ "icon" }` item?

**Reply by pynappo** [reply]
you'll need to copy the entire file table:

```lua
require('neo-tree').setup({
  renderers = {
    file = {
      { "indent" },
      -- { "icon" },
      {
        "container",
        content = {
          {
            "name",
            zindex = 10
          },
          {
            "symlink_target",
            zindex = 10,
            highlight = "NeoTreeSymbolicLinkTarget",
          },
          { "clipboard", zindex = 10 },
          { "bufnr", zindex = 10 },
          { "modified", zindex = 20, align = "right" },
          { "diagnostics",  zindex = 20, align = "right" },
          { "git_status", zindex = 10, align = "right" },
          { "file_size", zindex = 10, align = "right" },
          { "type", zindex = 10, align = "right" },
          { "last_modified", zindex = 10, align = "right" },
          { "created", zindex = 10, align = "right" },
        },
      },
    },
})
```

**Reply by ChrisTaylorDeveloper** [reply]
Many thanks.
